akka-http/src/main/scala/akka/http/scaladsl/marshalling/PredefinedToResponseMarshallers.scala

```diff
@@ -117,18 +93,20 @@ trait LowPriorityToResponseMarshallerImplicits {
   implicit def liftMarshaller[T](implicit m: ToEntityMarshaller[T]): ToResponseMarshaller[T] =
     PredefinedToResponseMarshallers.fromToEntityMarshaller()
 
-  // FIXME deduplicate this!!!
-  implicit def fromEntityStreamingSupportAndEntityMarshaller[T, M](implicit s: EntityStreamingSupport, m: ToEntityMarshaller[T], tag: ClassTag[T]): ToResponseMarshaller[Source[T, M]] = {
+  implicit def fromEntityStreamingSupportAndEntityMarshaller[T, M](implicit s: EntityStreamingSupport, m: ToEntityMarshaller[T], tag: ClassTag[T]): ToResponseMarshaller[Source[T, M]] =
+    fromEntityStreamingSupportAndByteStringSourceMarshaller[T, M](s, m.map(_.dataBytes))
+
+  private[marshalling] def fromEntityStreamingSupportAndByteStringSourceMarshaller[T: ClassTag, M](s: EntityStreamingSupport, m: Marshaller[T, Source[ByteString, _]]): ToResponseMarshaller[Source[T, M]] = {
     Marshaller[Source[T, M], HttpResponse] { implicit ec ⇒ source ⇒
       FastFuture successful {
         Marshalling.WithFixedContentType(s.contentType, () ⇒ {
           val availableMarshallingsPerElement = source.mapAsync(1) { t ⇒ m(t)(ec) }
 
           // TODO optimise such that we pick the optimal marshalling only once (headAndTail needed?)
           // TODO, NOTE: this is somewhat duplicated from Marshal.scala it could be made DRYer
-          val bestMarshallingPerElement = availableMarshallingsPerElement mapConcat { marshallings ⇒
+          val bestMarshallingPerElement = availableMarshallingsPerElement map { marshallings ⇒
```

Comment:
The rest of the changes here should just simplify the code a bit (unless I missed something about why it was the way it was before).

Reply:
This might have been influenced by the 'somewhat duplicated' code in Marshal.scala? Did you look at the test coverage for this feature, did it seem sufficient to give confidence the behavior didn't change?

Reply:
As far as I can tell, this code is covered by `EntityStreamingSpec.scala`.